programs/lib/Makefrag-lib

```diff
@@ -6,16 +6,25 @@ LIB_SOURCES += \
 	$(FP_LIB_DIR)/flexpret_exceptions.c 
 else
 LIB_SOURCES += \
-	$(FP_LIB_DIR)/printf/src/printf/printf.c \
 	$(FP_LIB_DIR)/syscalls/syscalls.c \
 	$(FP_LIB_DIR)/syscalls/printf_putchar.c \
 	$(FP_LIB_DIR)/flexpret_thread.c \
 	$(FP_LIB_DIR)/flexpret_lock.c \
 	$(FP_LIB_DIR)/flexpret_cond.c \
-	$(FP_LIB_DIR)/tinyalloc/tinyalloc.c \
-	$(FP_LIB_DIR)/flexpret_exceptions.c 
+	$(FP_LIB_DIR)/flexpret_exceptions.c \
+	$(FP_LIB_DIR)/ctx_switch.S
+
+ifeq ($(PRINTF_ENABLED),true)
+LIB_SOURCES += $(FP_LIB_DIR)/printf/src/printf/printf.c
+LIB_INCS += 
+else
+LIB_SOURCES += $(FP_LIB_DIR)/syscalls/printf_stubs.c
```

Comment:
`printf_stubs.c` just has empty implementations for the printf-familiy of functions.